java/client/src/main/java/glide/api/OpenTelemetry.java

```
@@ -0,0 +1,424 @@
+     * Initializes the OpenTelemetry integration with the provided configuration. This method should
+     * be called before any Valkey GLIDE client operations to enable OpenTelemetry tracing and metrics
+     * collection. If OpenTelemetry is already initialized, this method will log a warning and take no
+     * action.
+     *
+     * @param config The OpenTelemetry configuration containing settings for traces and metrics
+     * @example
+     *     <pre>{@code
+     * import glide.api.OpenTelemetry;
+     * OpenTelemetry.init(
+     *      OpenTelemetry.OpenTelemetryConfig.builder()
+     *         .traces(
+     *             OpenTelemetry.TracesConfig.builder()
+     *                 .endpoint("http://localhost:4318/v1/traces")
+     *                 .samplePercentage(10) // Optional, defaults to 1. Can also be changed at runtime via setSamplePercentage().
+     *                 .build()
+     *          )
+     *          .metrics(
+     *             OpenTelemetry.MetricsConfig.builder()
+     *                 .endpoint("http://localhost:4318/v1/metrics")
+     *                 .build()
+     *          )
+     *         .flushIntervalMs(5000L) // Optional, defaults to 5000
+     *         .build()
+     * );
+     * }</pre>
+     */
+    public static void init(OpenTelemetryConfig config) {
+        if (instance == null) {
+            internalInit(config);
+            Logger.log(Logger.Level.INFO, "GlideOpenTelemetry", "OpenTelemetry initialized");
+            return;
+        }
+
+        Logger.log(Logger.Level.WARN, "GlideOpenTelemetry", "OpenTelemetry already initialized");
+    }
+
+    private static void internalInit(OpenTelemetryConfig config) {
+        openTelemetryConfig = config;
+
+        String tracesEndpoint = null;
+        int tracesSamplePercentage = -1;
+        if (config.getTraces() == null && config.getMetrics() == null) {
+            Logger.log(Logger.Level.INFO, "GlideOpenTelemetry", "OpenTelemetry config error");
```

Maybe tell what is the error?
Maybe throw an exception or return `false`? If user suppresses logs (or doesn't monitor them in RT), it will think that otel is configured.

---

Updated. Added an exception.